.github/workflows/build-and-test.yml

```diff
@@ -49,3 +49,48 @@ jobs:
       coveragereports: PAWS.xcresult
     secrets:
       token: ${{ secrets.CODECOV_TOKEN }}
+  buildandtestdatapipelinenotebook:
+    name: Build and Test ECG Data pipeline Notebook
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up Python
+      uses: actions/setup-python@v5
+    - name: Setup NodeJS
+      uses: actions/setup-node@v3
+    - name: Setup Java
+      uses: actions/setup-java@v3
+      with:
+        distribution: 'microsoft'
+        java-version: '17'
+    - name: Setup LaTex
+      run: |
+        sudo apt-get install pandoc texlive-xetex texlive-fonts-recommended texlive-plain-generic
+    - name: Cache Firebase Emulators
+      uses: actions/cache@v3
+      with:
+        path: ~/.cache/firebase/emulators
+        key: ${{ runner.os }}-${{ runner.arch }}-firebase-emulators-${{ hashFiles('~/.cache/firebase/emulators/**') }}
+    - name: Install Firebase CLI Tools
+      run: npm install -g firebase-tools
+    - name: Install Infrastructure
+      run: |
+        python -m pip install --upgrade pip
+        pip install jupyterlab
+    - name: Install ECGDataPipelineTemplate Dependencies
+      run: |
+        pip install pandas numpy matplotlib firebase-admin requests ipywidgets pytz
+    - name: Set Firestore Emulator Environment Variable
+      run: |
+        echo "FIRESTORE_EMULATOR_HOST=localhost:8080" >> $GITHUB_ENV
+        echo "GCLOUD_PROJECT=spezidatapipelinetemplate" >> $GITHUB_ENV
```

Comment:
I suspect that google cloud project should be something closer to PAWs or ECG measurements?